src/stream_branches.py

```diff
@@ -453,7 +543,7 @@ def remove_branches_without_catchments(
         unique_catchments = set(catchments.loc[:, reach_id_attribute_in_catchments].unique())
 
         current_index_name = self.index.name
-        self.set_index(branch_id_attribute, drop=False, inplace=True)
+        self = self.set_index(branch_id_attribute, drop=False)
```

Comment:
This isn't critical, but I'm wondering if setting self and then returning self within the new `set_index` method, and then setting self again during this call is redundant. In other words, since you re-wrote the `set_index` method, could we just leave off the `self =` part of this line, now?

Reply:
@CarsonPruitt-NOAA Good question. I just tested it and `self` isn't updated without the assignment. Here's some sample output that illustrates that:
```
> self
           ID        to  order_  Lake            gages    Slope  ...     From_Node       To_Node  outlet_id inlet_id arbolate_sum    levpa_id
0    12297182  12297144       5 -9999                   0.00050  ...  122900000001  122900000002         -1        1     4.009258  1229000009
1    12297144  12297138       5 -9999                   0.00009  ...  122900000002  122900000003         -1       -1     7.805751  1229000009
2    12297138  12297136       5 -9999                   0.00001  ...  122900000003  122900000004         -1       -1     8.435007  1229000009
3    12297082  12297176       5 -9999                   0.00001  ...  122900000005  122900000006         -1       -1    41.399222  1229000003
4    12297078  12297082       5 -9999                   0.00005  ...  122900000007  122900000005         -1       -1    40.510577  1229000003
..        ...       ...     ...   ...              ...      ...  ...           ...           ...        ...      ...          ...         ...
510  12473978  12473996       5 -9999                   0.00119  ...  124700000514  124700000523        511       -1     7.518424  1247000012
511  12474010  12474024       5 -9999                   0.00099  ...  124700000524  124700000525        512      512     3.893549  1247000013
512  12471980  12471982       3 -9999                   0.00270  ...  124700000526  124700000527        513      513    13.804356  1247000014
513  12520753  12520765       4 -9999                   0.00088  ...  125200000484  125200000528        514       -1    21.788822  1252000015
514  12520903  12520907       3 -9999                   0.00703  ...  125200000529  125200000530        515      515     1.409630  1252000016

[515 rows x 16 columns]

> self.index
RangeIndex(start=0, stop=515, step=1)

> self.set_index(branch_id_attribute, drop=False)
                  ID        to  order_  Lake            gages    Slope  ...     From_Node       To_Node  outlet_id inlet_id arbolate_sum    levpa_id
levpa_id                                                                ...                                                                         
1229000009  12297182  12297144       5 -9999                   0.00050  ...  122900000001  122900000002         -1        1     4.009258  1229000009
1229000009  12297144  12297138       5 -9999                   0.00009  ...  122900000002  122900000003         -1       -1     7.805751  1229000009
1229000009  12297138  12297136       5 -9999                   0.00001  ...  122900000003  122900000004         -1       -1     8.435007  1229000009
1229000003  12297082  12297176       5 -9999                   0.00001  ...  122900000005  122900000006         -1       -1    41.399222  1229000003
1229000003  12297078  12297082       5 -9999                   0.00005  ...  122900000007  122900000005         -1       -1    40.510577  1229000003
...              ...       ...     ...   ...              ...      ...  ...           ...           ...        ...      ...          ...         ...
1247000012  12473978  12473996       5 -9999                   0.00119  ...  124700000514  124700000523        511       -1     7.518424  1247000012
1247000013  12474010  12474024       5 -9999                   0.00099  ...  124700000524  124700000525        512      512     3.893549  1247000013
1247000014  12471980  12471982       3 -9999                   0.00270  ...  124700000526  124700000527        513      513    13.804356  1247000014
1252000015  12520753  12520765       4 -9999                   0.00088  ...  125200000484  125200000528        514       -1    21.788822  1252000015
1252000016  12520903  12520907       3 -9999                   0.00703  ...  125200000529  125200000530        515      515     1.409630  1252000016

[515 rows x 16 columns]

> self
           ID        to  order_  Lake            gages    Slope  ...     From_Node       To_Node  outlet_id inlet_id arbolate_sum    levpa_id
0    12297182  12297144       5 -9999                   0.00050  ...  122900000001  122900000002         -1        1     4.009258  1229000009
1    12297144  12297138       5 -9999                   0.00009  ...  122900000002  122900000003         -1       -1     7.805751  1229000009
2    12297138  12297136       5 -9999                   0.00001  ...  122900000003  122900000004         -1       -1     8.435007  1229000009
3    12297082  12297176       5 -9999                   0.00001  ...  122900000005  122900000006         -1       -1    41.399222  1229000003
4    12297078  12297082       5 -9999                   0.00005  ...  122900000007  122900000005         -1       -1    40.510577  1229000003
..        ...       ...     ...   ...              ...      ...  ...           ...           ...        ...      ...          ...         ...
510  12473978  12473996       5 -9999                   0.00119  ...  124700000514  124700000523        511       -1     7.518424  1247000012
511  12474010  12474024       5 -9999                   0.00099  ...  124700000524  124700000525        512      512     3.893549  1247000013
512  12471980  12471982       3 -9999                   0.00270  ...  124700000526  124700000527        513      513    13.804356  1247000014
513  12520753  12520765       4 -9999                   0.00088  ...  125200000484  125200000528        514       -1    21.788822  1252000015
514  12520903  12520907       3 -9999                   0.00703  ...  125200000529  125200000530        515      515     1.409630  1252000016

[515 rows x 16 columns]

> self.index
RangeIndex(start=0, stop=515, step=1)
```

After assigning `self = self.set_index()`:
```
> self = self.set_index(branch_id_attribute, drop=False)
> self
                  ID        to  order_  Lake            gages    Slope  ...     From_Node       To_Node  outlet_id inlet_id arbolate_sum    levpa_id
levpa_id                                                                ...                                                                         
1229000009  12297182  12297144       5 -9999                   0.00050  ...  122900000001  122900000002         -1        1     4.009258  1229000009
1229000009  12297144  12297138       5 -9999                   0.00009  ...  122900000002  122900000003         -1       -1     7.805751  1229000009
1229000009  12297138  12297136       5 -9999                   0.00001  ...  122900000003  122900000004         -1       -1     8.435007  1229000009
1229000003  12297082  12297176       5 -9999                   0.00001  ...  122900000005  122900000006         -1       -1    41.399222  1229000003
1229000003  12297078  12297082       5 -9999                   0.00005  ...  122900000007  122900000005         -1       -1    40.510577  1229000003
...              ...       ...     ...   ...              ...      ...  ...           ...           ...        ...      ...          ...         ...
1247000012  12473978  12473996       5 -9999                   0.00119  ...  124700000514  124700000523        511       -1     7.518424  1247000012
1247000013  12474010  12474024       5 -9999                   0.00099  ...  124700000524  124700000525        512      512     3.893549  1247000013
1247000014  12471980  12471982       3 -9999                   0.00270  ...  124700000526  124700000527        513      513    13.804356  1247000014
1252000015  12520753  12520765       4 -9999                   0.00088  ...  125200000484  125200000528        514       -1    21.788822  1252000015
1252000016  12520903  12520907       3 -9999                   0.00703  ...  125200000529  125200000530        515      515     1.409630  1252000016

[515 rows x 16 columns]

> self.index
Index(['1229000009', '1229000009', '1229000009', '1229000003', '1229000003',
       '1229000003', '1229000003', '1229000003', '1229000021', '1229000021',
       ...
       '1231000007', '1230000008', '1229000009', '1253000010', '1253000011',
       '1247000012', '1247000013', '1247000014', '1252000015', '1252000016'],
      dtype='object', name='levpa_id', length=515)

```